src/Compilers/CSharp/Test/Semantic/Semantics/NullCheckedParameterTests.cs

```diff
@@ -114,25 +114,71 @@ public void NullCheckedPartialMethod()
             var source = @"
 partial class C
 {
-    partial void M(string x!);
+    partial void M(string x!!);
 }";
             CreateCompilation(source, parseOptions: TestOptions.RegularPreview).VerifyDiagnostics(
                     // (4,27): error CS8717: Parameter 'x' can only have exclamation-point null checking in implementation methods.
-                    //     partial void M(string x!);
+                    //     partial void M(string x!!);
                     Diagnostic(ErrorCode.ERR_MustNullCheckInImplementation, "x").WithArguments("x").WithLocation(4, 27));
         }
 
+        [Fact]
+        public void NullCheckedBadSyntax()
+        {
+            var source = @"
+partial class C
+{
+    void M0(string name !!=null) { }
```

Comment:
These declarations need to be in the parsing tests, one test per method declaration, so we can assert the shape of the tree being produced.

Reply:
Added a bunch of parsing tests.